hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoDeleteTransitionLogic.java

```diff
@@ -61,46 +64,59 @@
 
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
-	private final HederaLedger ledger;
 	private final TransactionContext txnCtx;
+	private final AccountStore accountStore;
+	private final TypedTokenStore tokenStore;
+	private final TransactionRecordService recordService;
 
 	@Inject
-	public CryptoDeleteTransitionLogic(HederaLedger ledger, TransactionContext txnCtx) {
-		this.ledger = ledger;
+	public CryptoDeleteTransitionLogic(TransactionContext txnCtx,
+									   AccountStore accountStore,
+									   TypedTokenStore tokenStore,
+									   TransactionRecordService recordService) {
 		this.txnCtx = txnCtx;
+		this.accountStore = accountStore;
+		this.tokenStore = tokenStore;
+		this.recordService = recordService;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			CryptoDeleteTransactionBody op = txnCtx.accessor().getTxn().getCryptoDelete();
+		/* --- Translate from gRPC types --- */
+		final var op = txnCtx.accessor().getTxn().getCryptoDelete();
 
-			AccountID id = op.getDeleteAccountID();
-			if (ledger.isKnownTreasury(id)) {
-				txnCtx.setStatus(ACCOUNT_IS_TREASURY);
-				return;
-			}
-			AccountID beneficiary = op.getTransferAccountID();
-			if (ledger.isDetached(id) || ledger.isDetached(beneficiary)) {
-				txnCtx.setStatus(ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
-				return;
-			}
+		final var subjectId = Id.fromGrpcAccount(op.getDeleteAccountID());
+		final var beneficiaryId = Id.fromGrpcAccount(op.getTransferAccountID());
 
-			if (!ledger.allTokenBalancesVanish(id)) {
-				txnCtx.setStatus(TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
-				return;
-			}
+		final var subject = accountStore.loadAccount(subjectId);
+		final var beneficiary = accountStore.loadAccount(beneficiaryId);
+
+		/* --- Validate --- */
+		validateFalse(tokenStore.isKnownTreasury(subjectId), ACCOUNT_IS_TREASURY);
+		validateFalse(subject.isSmartContract(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
+		validateFalse(beneficiary.isSmartContract(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
+		validateAllTokenBalancesVanish(subject);
 
-			ledger.delete(id, beneficiary);
+		/* --- Do the business logic --- */
+		final List<BalanceChange> balanceChanges = subject.transferHbar(beneficiary, subject.getBalance());
+		subject.setDeleted(true);
```

Comment:
Let's define a `delete` method on the `Account` class, which executes this bussiness logic ( the setter )